api/listener.go

```diff
@@ -95,27 +96,32 @@ func (cl *chainListener) ReceiveBlock(blk *block.Block) error {
 }
 
 // AddResponder adds a new responder
-func (cl *chainListener) AddResponder(responder apitypes.Responder) (listenerID string, err error) {
-	if cl.streamMap.Count() >= cl.capacity {
-		err = errorCapacityReached
-		return
-	}
+func (cl *chainListener) NewStreamID() (listenerID string) {
 	i := 0
-	for ; i < idRetry; i++ {
+	for ; i < _idRetry; i++ {
 		listenerID = cl.idGenerator.newID()
 		if _, exist := cl.streamMap.Get(listenerID); !exist {
 			break
 		}
 	}
-	if i == idRetry {
-		err = errNoID
-		return
+	if i == _idRetry {
+		// An new id is assumed to be found, because comninations (2^62) is far larger than capacity
+		log.L().Panic("No peer id available", zap.Error(errNoID))
+		panic(errNoID)
```

Comment:
1. comninations -> combinations
2. just `log.L().Panic("No peer id available")`, no need to define `errNoID` then
3. no need to `panic()` again,  `log.L().Panic` calls panic inside